Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift

```diff
@@ -55,15 +56,20 @@ class DocumentViewModel: ObservableObject {
 
 extension DocumentViewModel {
     func addAnnotationIfWithinBounds(center: CGPoint, bounds: CGRect) {
+        guard let addedSelectionBox = addedSelectionBox else {
+            // If we click outside the document, there will not be a selection box
+            return
+        }
         guard let currentUser = AnnotatoAuth().currentUser else {
+            addedSelectionBox.didDelete()
             return
         }
-
         let newAnnotationWidth = 300.0
         let newAnnotation = Annotation(
             origin: .zero,
             width: newAnnotationWidth,
             parts: [],
+            selectionBox: addedSelectionBox.model,
```

Comment:
If the annotation model contains selectionBox model, why is the comment from lines 87 to 92 relevant? When the annotationViewModel binds to annotationModel, wouldnt it also bind to the selectionBoxViewModel?

I think if you follow the above comment where the view model is in charge of making the entire annotation-selectionbox set at the same time, there isn't the problem where the view creates the selection box first followed by the annotation, causing a need for a separate selection box view model first.

Reply:
I think that the document view model needs to create the selection box first before the annotation because ideally I would like the user to see his selection box as he pans his finger on the screen (and to do that I need to already add a selection box, and the documentview needs to be updated already), then only when he releases his finger then the annotation gets added (which requires another function call)